### ISSUE TITLE
[LC_CTRL/rtl] token_if_fsm_err_i prevents LC_CTRL FSM entering escalate.

### ISSUE THREAD

**Issue by NigelScales** [lowRISC/opentitan#12204]
To reproduce use PR https://github.com/lowRISC/opentitan/pull/12203
./util/dvsim/dvsim.py hw/ip/lc_ctrl/dv/lc_ctrl_sim_cfg.hjson -i lc_ctrl_state_failure --fixed-seed 2184973054 --waves -mp 3 --tool xcelium -v l

While fixing the backdoor read and write for the KMAC I/F FSM state vector and  it appears that the error output from the KMAC FSM 
sent to the LC_CTRL FSM as token_if_fsm_err_i prevents entering EscalateSt as a result of an assertion  of the scrap inputs.

Syndrome:
UVM_ERROR @   8136033 ps: (lc_ctrl_scoreboard.sv:327) [uvm_test_top.env.scoreboard] Check failed csr.get_mirrored_value() == item.d_data (762010326 [0x2d6b5ad6] vs 796647159 [0x2f7bdef7]) reg name: lc_ctrl_reg_block.lc_state

```
    // If at any time the life cycle state encoding or any other FSM state within this module
    // is not valid, we jump into the terminal error state right away.
    // Note that state_invalid_error is a multibit error signal
    // with different error sources - need to reduce this to one bit here.
    // SEC_CM: MAIN.FSM.LOCAL_ESC
    if (|state_invalid_error | token_if_fsm_err_i) begin
      fsm_state_d = InvalidSt;
      state_invalid_error_o = 1'b1;
    // SEC_CM: MAIN.FSM.GLOBAL_ESC
    end else if (esc_scrap_state0_i || esc_scrap_state1_i) begin
      fsm_state_d = EscalateSt;
    end
```

![image](https://user-images.githubusercontent.com/78102677/164046303-c6e4c8fa-f0ca-4471-9bb8-4210909fe6ba.png)

**Comment by msfschaffner** [comment]
@tjaychen should we reverse the priority in this if statement so that escalation can override `InvalidSt`? 

I believe initially when I wrote this I did it this way so that you can tell via the LC TAP whether the system has shut down due to either a failure inside LC_CTRL, or due to global escalation. However, going through the `InvalidSt` will also set the `state_invalid_error_o`, so we should still be able to tell, even if the final state is `EscalateSt`... WDYT?

**Comment by tjaychen** [comment]
wait sorry, i think i'm not completely following.. but in the case we do as you say and go to `EscalateSt`, how will we be able to tell there was a state invalid error..? Do you mean the tap can simply read back the error conditions?

**Comment by msfschaffner** [comment]
Yes, the error bits are latched here (and they are readable via the TAP): https://docs.opentitan.org/hw/ip/lc_ctrl/doc/index.html#Reg_status

**Comment by tjaychen** [comment]
sounds good to me! I guess it does make sense for global escalation to have
priority..especially if all the status is discoverable.

**Comment by msfschaffner** [comment]
yeah and thinking about it some more, it actually allows us to tell from the outside whether the system has actually escalated or not (although we could also read the reset cause upon the next boot).

**Comment by NigelScales** [comment]
@msfschaffner  Not fixed due to typo.
See https://github.com/lowRISC/opentitan/pull/12218

**Comment by NigelScales** [comment]
PR https://github.com/lowRISC/opentitan/pull/12228 has fixed this issue.
